almkanal/almkanal.py

```diff
@@ -0,0 +1,629 @@
+from pathlib import Path
+
+import attrs
+import mne
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from numpy.typing import ArrayLike, NDArray
+
+from almkanal.data_utils.check_data import check_raw_epoch
+from almkanal.data_utils.data_classes import ICAInfoDict, InfoClass, PickDictClass
+from almkanal.preproc_utils.bio_utils import run_bio_preproc
+from almkanal.preproc_utils.ica_utils import run_ica
+
+# all them utility functions
+from almkanal.preproc_utils.maxwell_utils import run_maxwell
+from almkanal.src_utils.headmodel_utils import compute_headmodel, make_fwd
+from almkanal.src_utils.src_utils import comp_spatial_filters, src2parc
+
+
+@define
+class AlmKanal:
+    raw: None | mne.io.Raw = None
+    epoched: None | mne.Epochs = None
```

Comment:
I think, a composition based approach would make this class easier to understand and use. For instance, define a `SourceProjection` class that contains `fwd` and `filter` and have `Almkanal` have a property that is an object of this `SourceProjection` class. Same for ICA and anything TRF related.